From 7d65765bf341a218369497618d790c64b0072ec6 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Mon, 26 Feb 2018 14:49:21 +0800
Subject: [PATCH] arm64: dts: rockchip: Enable SD/SDIO 3.0 for PX30/RK3326
 boards

This patch enables SD(IO) 3.0 for all boards by adding correct
vccq and vcc power supplies, as well as properities required by
UHS-I mode.

Change-Id: Iec11e1d1abe7ef9fc17ba08eece3440d7dcaea0b
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 .../boot/dts/rockchip/px30-evb-ddr3-v10.dts   | 16 +++++++++-----
 .../boot/dts/rockchip/px30-evb-ddr4-v10.dts   | 14 +++++++++----
 arch/arm64/boot/dts/rockchip/px30.dtsi        | 21 ++++++++++---------
 .../boot/dts/rockchip/rk3326-evb-lp3-v10.dts  | 14 +++++++++----
 4 files changed, 42 insertions(+), 23 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
index 97d4cc8af526..138ab8a8a3d6 100644
--- a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dts
@@ -445,7 +445,7 @@
 			vccio_sd: LDO_REG5 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <3300000>;
+				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
 
 				regulator-name = "vccio_sd";
@@ -649,25 +649,31 @@
 };
 
 &sdmmc {
+	bus-width = <4>;
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
 	supports-sd;
-	broken-cd;
 	card-detect-delay = <800>;
 	ignore-pm-notify;
-	keep-power-in-suspend;
 	/*cd-gpios = <&gpio2 4 GPIO_ACTIVE_HIGH>; [> CD GPIO <]*/
-	status = "disabled";
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	vqmmc-supply = <&vccio_sd>;
+	vmmc-supply = <&vcc_sd>;
+	status = "okay";
 };
 
 &sdio {
-	max-frequency = <50000000>;
+	bus-width = <4>;
 	cap-sd-highspeed;
 	supports-sdio;
 	ignore-pm-notify;
 	keep-power-in-suspend;
 	non-removable;
 	mmc-pwrseq = <&sdio_pwrseq>;
+	sd-uhs-sdr104;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts b/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
index 089dd0d166ef..65ee0fb1ad7e 100644
--- a/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/px30-evb-ddr4-v10.dts
@@ -446,7 +446,7 @@
 			vccio_sd: LDO_REG5 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <3300000>;
+				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
 
 				regulator-name = "vccio_sd";
@@ -649,25 +649,31 @@
 };
 
 &sdmmc {
+	bus-width = <4>;
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
 	supports-sd;
-	broken-cd;
 	card-detect-delay = <800>;
 	ignore-pm-notify;
-	keep-power-in-suspend;
 	/*cd-gpios = <&gpio2 4 GPIO_ACTIVE_HIGH>; [> CD GPIO <]*/
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	vqmmc-supply = <&vccio_sd>;
+	vmmc-supply = <&vcc_sd>;
 	status = "disabled";
 };
 
 &sdio {
-	max-frequency = <50000000>;
+	bus-width = <4>;
 	cap-sd-highspeed;
 	supports-sdio;
 	ignore-pm-notify;
 	keep-power-in-suspend;
 	non-removable;
 	mmc-pwrseq = <&sdio_pwrseq>;
+	sd-uhs-sdr104;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/px30.dtsi b/arch/arm64/boot/dts/rockchip/px30.dtsi
index 9fd58103ad89..da432b6902f0 100644
--- a/arch/arm64/boot/dts/rockchip/px30.dtsi
+++ b/arch/arm64/boot/dts/rockchip/px30.dtsi
@@ -977,8 +977,9 @@
 		compatible = "rockchip,px30-dw-mshc", "rockchip,rk3288-dw-mshc";
 		reg = <0x0 0xff370000 0x0 0x4000>;
 		max-frequency = <150000000>;
-		clocks = <&cru HCLK_SDMMC>, <&cru SCLK_SDMMC>;
-		clock-names = "biu", "ciu";
+		clocks = <&cru HCLK_SDMMC>, <&cru SCLK_SDMMC>,
+			 <&cru SCLK_SDMMC_DRV>, <&cru SCLK_SDMMC_SAMPLE>;
+		clock-names = "biu", "ciu", "ciu-drv", "ciu-sample";
 		power-domains = <&power PX30_PD_SDCARD>;
 		fifo-depth = <0x100>;
 		interrupts = <GIC_SPI 54 IRQ_TYPE_LEVEL_HIGH>;
@@ -2078,30 +2079,30 @@
 		sdmmc {
 			sdmmc_clk: sdmmc-clk {
 				rockchip,pins =
-					<1 RK_PD6 RK_FUNC_1 &pcfg_pull_none_4ma>;
+					<1 RK_PD6 RK_FUNC_1 &pcfg_pull_none_8ma>;
 			};
 
 			sdmmc_cmd: sdmmc-cmd {
 				rockchip,pins =
-					<1 RK_PD7 RK_FUNC_1 &pcfg_pull_up_4ma>;
+					<1 RK_PD7 RK_FUNC_1 &pcfg_pull_up_8ma>;
 			};
 
 			sdmmc_det: sdmmc-det {
 				rockchip,pins =
-					<0 RK_PA3 RK_FUNC_1 &pcfg_pull_up_4ma>;
+					<0 RK_PA3 RK_FUNC_1 &pcfg_pull_up_8ma>;
 			};
 
 			sdmmc_bus1: sdmmc-bus1 {
 				rockchip,pins =
-					<1 RK_PD2 RK_FUNC_1 &pcfg_pull_up_4ma>;
+					<1 RK_PD2 RK_FUNC_1 &pcfg_pull_up_8ma>;
 			};
 
 			sdmmc_bus4: sdmmc-bus4 {
 				rockchip,pins =
-					<1 RK_PD2 RK_FUNC_1 &pcfg_pull_up_4ma>,
-					<1 RK_PD3 RK_FUNC_1 &pcfg_pull_up_4ma>,
-					<1 RK_PD4 RK_FUNC_1 &pcfg_pull_up_4ma>,
-					<1 RK_PD5 RK_FUNC_1 &pcfg_pull_up_4ma>;
+					<1 RK_PD2 RK_FUNC_1 &pcfg_pull_up_8ma>,
+					<1 RK_PD3 RK_FUNC_1 &pcfg_pull_up_8ma>,
+					<1 RK_PD4 RK_FUNC_1 &pcfg_pull_up_8ma>,
+					<1 RK_PD5 RK_FUNC_1 &pcfg_pull_up_8ma>;
 			};
 
 			sdmmc_gpio: sdmmc-gpio {
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
index 95559913a447..2cab67df0111 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10.dts
@@ -473,7 +473,7 @@
 			vccio_sd: LDO_REG5 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <3300000>;
+				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
 
 				regulator-name = "vccio_sd";
@@ -736,25 +736,31 @@
 };
 
 &sdmmc {
+	bus-width = <4>;
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
 	supports-sd;
-	broken-cd;
 	card-detect-delay = <800>;
 	ignore-pm-notify;
-	keep-power-in-suspend;
 	/*cd-gpios = <&gpio2 4 GPIO_ACTIVE_HIGH>; [> CD GPIO <]*/
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	vqmmc-supply = <&vccio_sd>;
+	vmmc-supply = <&vcc_sd>;
 	status = "disabled";
 };
 
 &sdio {
-	max-frequency = <50000000>;
+	bus-width = <4>;
 	cap-sd-highspeed;
 	supports-sdio;
 	ignore-pm-notify;
 	keep-power-in-suspend;
 	non-removable;
 	mmc-pwrseq = <&sdio_pwrseq>;
+	sd-uhs-sdr104;
 	status = "okay";
 };
 
-- 
2.35.3

